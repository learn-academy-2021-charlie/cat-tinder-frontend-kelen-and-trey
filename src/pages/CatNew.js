import React, { Component } from 'react'
import { Button, Form, Label, Input, FormGroup } from 'reactstrap'
import { Redirect } from 'react-router-dom'

class CatNew extends Component{
  constructor(props){
    super(props)
    this.state = {
      cat: {
        name: '',
        age: '',
        enjoys: ''
      },
      submitted: false,
    }
  }

  handleChange = e => {
    let form = this.state.cat
    form[e.target.name] = e.target.value
    this.setState({cat: form})
  }

  handleSubmit = () => {
    this.props.handleSubmit(this.state.cat)
    this.setState({submitted:true})
  }

  render(){
    return(
      <>
        <h2>CatNew Form Component</h2>
        <Form>
          <FormGroup>
            <Label for="name">Name</Label>
            <Input name="name" id="nameNew" onChange={this.handleChange}/>
          </FormGroup>
          <FormGroup>
            <Label for="age">Age</Label>
            <Input type="number" name="age" id="ageNew" onChange={this.handleChange}/>
          </FormGroup>
          <FormGroup>
            <Label for="enjoys">Enjoys</Label>
            <Input name="enjoys" id="enjoysNew" onChange={this.handleChange}/>
          </FormGroup>
          <Button onClick={this.handleSubmit}>Submit</Button>
        </Form>
        {this.state.submitted && <Redirect to="/catindex"/>}
      </>
    )
  }
}

export default CatNew
