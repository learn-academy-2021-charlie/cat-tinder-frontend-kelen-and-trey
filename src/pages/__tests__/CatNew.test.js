import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import cats from '../../mockCats'
import CatNew from '../CatNew'

Enzyme.configure({ adapter: new Adapter() })

describe('new cat form', () => {
  let renderedCatNew, mockSubmit
  beforeEach(() => {
    mockSubmit = jest.fn()
    renderedCatNew = shallow(<CatNew handleSubmit={mockSubmit}/> )
  })
  it('updates the name when given a name event', () => {
    expect(renderedCatNew).toMatchSnapshot()
    const mockEvent = {
      target: {
        name: 'name',
        value: 'freedom',
      }
    }

    const expected = {
      cat: {
        name: 'freedom',
        age: '',
        enjoys: '',
      },
      submitted: false,
    }

    renderedCatNew.instance().handleChange(mockEvent)
    expect(renderedCatNew.state()).toEqual(expected)
  })
  it('updates the age when given an age event', () => {
    expect(renderedCatNew).toMatchSnapshot()
    const mockEvent = {
      target: {
        name: 'age',
        value: '10',
      }
    }

    const expected = {
      cat: {
        name: '',
        age: '10',
        enjoys: '',
      },
      submitted: false,
    }

    renderedCatNew.instance().handleChange(mockEvent)
    expect(renderedCatNew.state()).toEqual(expected)
  })
  it('updates the enjoys value when given an enjoys event', () => {
    expect(renderedCatNew).toMatchSnapshot()
    const mockEvent = {
      target: {
        name: 'enjoys',
        value: 'lying down',
      }
    }

    const expected = {
      cat: {
        name: '',
        age: '',
        enjoys: 'lying down',
      },
      submitted: false,
    }

    renderedCatNew.instance().handleChange(mockEvent)
    expect(renderedCatNew.state()).toEqual(expected)
  })
  it("should call submit with the correct params", () => {
    renderedCatNew.setState({
      cat: {
        name: 'test name',
        age: '10',
        enjoys: 'test enjoys',
      },
      submitted: true
    });
    const expected = {
      name: 'test name',
      age: '10',
      enjoys: 'test enjoys',
    };
    const mockPreventDefault = jest.fn();
    const mockEvent = {
      preventDefault: mockPreventDefault
    };
    renderedCatNew.instance().handleSubmit(mockEvent);
    expect(mockSubmit).toHaveBeenCalledWith(expected);
  });
})
