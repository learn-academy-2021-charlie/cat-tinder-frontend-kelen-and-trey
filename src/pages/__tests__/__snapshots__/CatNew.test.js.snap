// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new cat form updates the age when given an age event 1`] = `ShallowWrapper {}`;

exports[`new cat form updates the age when the user types a number into the age field 1`] = `ShallowWrapper {}`;

exports[`new cat form updates the enjoys value when given an enjoys event 1`] = `ShallowWrapper {}`;

exports[`new cat form updates the name when given a name event 1`] = `ShallowWrapper {}`;

exports[`new cat form updates the name when the user types in the name field 1`] = `ShallowWrapper {}`;
